pkg/plugins/policies/meshtcproute/api/v1alpha1/validator.go

```diff
@@ -0,0 +1,104 @@
+package v1alpha1
+
+import (
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/validators"
+	matcher_validators "github.com/kumahq/kuma/pkg/plugins/policies/matchers/validators"
+)
+
+func (r *MeshTCPRouteResource) validate() error {
+	var verr validators.ValidationError
+
+	path := validators.RootedAt("spec")
+
+	verr.AddErrorAt(path.Field("targetRef"), validateTop(r.Spec.TargetRef))
+
+	if len(r.Spec.To) == 0 {
+		verr.AddViolationAt(path.Field("to"), "needs at least one item")
+	}
+
+	verr.AddErrorAt(path, validateTo(r.Spec.To))
+
+	return verr.OrNil()
+}
+
+func validateTop(targetRef common_api.TargetRef) validators.ValidationError {
+	return matcher_validators.ValidateTargetRef(targetRef, &matcher_validators.ValidateTargetRefOpts{
+		SupportedKinds: []common_api.TargetRefKind{
+			common_api.Mesh,
+			common_api.MeshSubset,
+			common_api.MeshService,
+			common_api.MeshServiceSubset,
+		},
+	})
+}
+
+func validateTo(to []To) validators.ValidationError {
+	var verr validators.ValidationError
+
+	for idx, toItem := range to {
+		path := validators.RootedAt("to").Index(idx)
+		verr.AddErrorAt(path.Field("targetRef"), matcher_validators.ValidateTargetRef(toItem.TargetRef, &matcher_validators.ValidateTargetRefOpts{
+			SupportedKinds: []common_api.TargetRefKind{
+				common_api.MeshService,
+			},
+		}))
+		verr.AddErrorAt(path.Field("rules"), validateRules(toItem.Rules))
+	}
+
+	return verr
+}
+
+func validateRules(rules []Rule) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if len(rules) != 1 {
```

Comment:
This is already validated by the kubebuilder annotations

Reply:
Good point: addressed